src/appleseed/foundation/image/genericprogressiveimagefilereader.cpp

```diff
@@ -262,59 +309,62 @@ Tile* GenericProgressiveImageFileReader::read_tile(
         // the correct dimensions, at some expenses.
         //
 
-        unique_ptr<Tile> source_tile(
-            new Tile(
-                impl->m_props.m_tile_width,
-                impl->m_props.m_tile_height,
-                impl->m_props.m_channel_count,
-                impl->m_props.m_pixel_format));
-
         const size_t origin_x = tile_x * impl->m_props.m_tile_width;
         const size_t origin_y = tile_y * impl->m_props.m_tile_height;
-
-        if (!impl->m_input->read_tile(
-                static_cast<int>(origin_x),
-                static_cast<int>(origin_y),
-                0, // z
-                impl->m_input->spec().format,
-                source_tile->get_storage()))
-            throw ExceptionIOError(impl->m_input->geterror().c_str());
-
         const size_t tile_width = min(impl->m_props.m_tile_width, impl->m_props.m_canvas_width - origin_x);
         const size_t tile_height = min(impl->m_props.m_tile_height, impl->m_props.m_canvas_height - origin_y);
 
+        // The tile fits perfectly into the canvas.
```

Comment:
Nice to have thought of that!

You should move the comment inside the `if`, it would be clearer.